packages/@lwc/engine/src/framework/base-lightning-element.ts

```diff
@@ -543,6 +549,38 @@ const baseDescriptors = ArrayReduce.call(
 
 defineProperties(BaseLightningElementConstructor.prototype, baseDescriptors);
 
+const ComponentConstructorAsCustomElementConstructorMap = new Map();
+
+function getCustomElementConstructor(Ctor: ComponentConstructor): HTMLElement {
+    if ((Ctor as any) === BaseLightningElementConstructor) {
```

Comment:
We shouldn't need the `as any` here. If it is needed it's because the argument type is wrong. Can we try to fix it?